src/core/datastore/sync/sync-manager.js

```diff
@@ -347,4 +352,66 @@ export class SyncManager {
         return Promise.reject(new SyncError('Invalid sync event name'));
     }
   }
+
+  _getPullLimit(pullOptions) {
+    const limitOption = pullOptions && pullOptions.pullLimit;
+    return limitOption || Infinity;
+  }
+
+  _getInternalPullQuery({ filter, sort, fields }, pullLimit) {
+    const query = new Query({ filter, sort, fields });
+    query.skip = 0;
+    query.limit = null; // the current default value
+
+    if (pullLimit && pullLimit < Infinity) {
+      query.limit = pullLimit;
+    }
+
+    return query;
+  }
+
+  _getPaginationQueries({ filter, sort, fields }, pageSize = maxEntityLimit, totalCount) {
+    const queryCount = Math.ceil(totalCount / pageSize);
+    const queries = times(queryCount, (i) => {
+      const query = new Query({ filter, sort, fields });
+      query.skip = i * pageSize;
+      query.limit = Math.min(totalCount - query.skip, pageSize);
+      return query;
+    });
+    return queries;
+  }
+
+  _fetchAndUpdateEntities(collection, query) {
+    return this._networkRepo.read(collection, query)
+      .then((entities) => {
+        if (isEmpty(entities)) {
+          return Promise.resolve();
+        }
+        return this._getOfflineRepo()
+          .then(repo => repo.update(collection, entities));
+      });
+  }
+
+  _executePaginationQueries(collection, deleteQuery, queries) {
+    return this._getOfflineRepo()
+      .then(repo => repo.delete(collection, deleteQuery))
+      .then(() => {
+        // TODO: throttle max request count - reuse from push?
+        return forEachAsync(queries, (query) => {
+          return this._fetchAndUpdateEntities(collection, query);
+        });
+      });
+  }
+
+  _paginate(collection, userQuery, options) {
+    const query = new Query({ filter: userQuery.filter }); // ignore sort, it's irrelevant
```

Comment:
It's only for the _count query. Other queries respect filter, sort and fields.

Perhaps it adds a tiny bit to the learning of how auto pagination works, but I don't think anyone would care much if their sort is missing from the _count request, especially since it doesn't make sense there, or maybe even causes a slowdown performance-wise (I'm not sure how backend treats this sort of query, or Mongo, but still).

That said, I don't feel very strongly either way - I will add back fields and sort, if you feel they are useful.